File: exercises/week01/part01/student.cpp
#include "student.h"

Student::Student() { }

Student::Student(std::string name, std::string student_id)
  : _name(name), _student_id(student_id), _credits(0) { }

Student::~Student() {
  printf("Student %s destructed.\n", _name.c_str());
}

//getters

std::string Student::name() { return _name; }

std::string Student::student_id() { return _student_id; }

int Student::credits(){ return _credits; }

//setters

void Student::set_name(std::string name) {
  _name = name;
}

void Student::set_student_id(std::string student_id){
  _student_id = student_id;
}

void Student::set_credits(int credits) {
  _credits = credits;
}

//public methods

void Student::print() {
  printf("Name: %s\nStudent number: %s\nCredits: %d\n", _name.c_str(), _student_id.c_str(), _credits);
}

File: exercises/week04/part02/Student.java
package part02;

public class Student {
  private String name;
  private String studentId;
  private int credits;
  private Address address;

  public Student() {
    System.out.println("Student class constructor.");
  }

  public Student(String name, String studentId) {
    this.name = name;
    this.studentId = studentId;

    System.out.println("Student class parameterized constructor.");
  }

  //getters

  public String getName() { return this.name; }
  public String getStudentId() { return this.studentId; }
  public int getCredits() {  return this.credits; }
  public Address getAddress() { return this.address; }

  //setters

  public void setName(String name) {
    this.name = name;
  }

  public void setStudentId(String studentId) {
    this.studentId = studentId;
  }

  public void setCredits(int credits) {
    this.credits = credits;
  }

  public void setAddress(Address address) {
    this.address = address;
  }

  //public methods

  public void print() {
    var info = String.format("Name: %s\nStudent number: %s\nCredits: %s", 
      this.getName(),
      this.getStudentId(),
      this.getCredits()
    );
    System.out.println(info);
  }
}
